### DISCUSSION TITLE
Question about the return value

### DISCUSSION THREAD

**Discussion by cymqqqq** [AluVM/aluvm#122] [Q&A]
I have some questions about the aluvm:

1. if I want to return a value in the aluvm script(whatever it is), what kind of instruction should I take? (Now there is a `ret` instr, but it returns nothing)
2. if I want to call another transition or extension, can I use `call` instr to call the lib id(hash) of a specific transition?

**Comment by dr-orlovsky** [comment]
In any registry-based assembler values are returned via registers. And there is no such thing as "calling other state transition or extensions": the question just doesn't make sense, like "how to call CNN from AluVM"
